tests/storage/hotplug.go

```diff
@@ -0,0 +1,623 @@
+			&expect.BSnd{S: "echo $?\n"},
+			&expect.BExp{R: console.RetValue("0")},
+			&expect.BSnd{S: fmt.Sprintf("sudo chmod a+w %s\n", filepath.Join("/test", filepath.Base(device), "data"))},
+			&expect.BExp{R: console.PromptExpression},
+			&expect.BSnd{S: "echo $?\n"},
+			&expect.BExp{R: console.RetValue("0")},
+			&expect.BSnd{S: fmt.Sprintf("echo '%s' > %s\n", vmi.UID, filepath.Join("/test", filepath.Base(device), "data/message"))},
+			&expect.BExp{R: console.PromptExpression},
+			&expect.BSnd{S: "echo $?\n"},
+			&expect.BExp{R: console.RetValue("0")},
+			&expect.BSnd{S: fmt.Sprintf("cat %s\n", filepath.Join("/test", filepath.Base(device), "data/message"))},
+			&expect.BExp{R: string(vmi.UID)},
+			&expect.BSnd{S: "sync\n"},
+			&expect.BExp{R: console.PromptExpression},
+			&expect.BSnd{S: "sync\n"},
+			&expect.BExp{R: console.PromptExpression},
+		}
+		Expect(console.SafeExpectBatch(vmi, batch, 20)).To(Succeed())
+	}
+
+	getTargetsFromVolumeStatus := func(vmi *kubevirtv1.VirtualMachineInstance, volumeNames ...string) []string {
+		nameMap := make(map[string]bool)
+		for _, volumeName := range volumeNames {
+			nameMap[volumeName] = true
+		}
+		res := make([]string, 0)
+		updatedVMI, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		for _, volumeStatus := range updatedVMI.Status.VolumeStatus {
+			if _, ok := nameMap[volumeStatus.Name]; ok && volumeStatus.HotplugVolume != nil {
+				Expect(volumeStatus.Target).ToNot(BeEmpty())
+				res = append(res, fmt.Sprintf("/dev/%s", volumeStatus.Target))
+			}
+		}
+		return res
+	}
+
+	Context("Offline VM", func() {
+		var (
+			vm *kubevirtv1.VirtualMachine
+		)
+		BeforeEach(func() {
+			By("Creating VirtualMachine")
+			template, _ := newVirtualMachineInstanceWithContainerDisk()
+			vm = createVirtualMachine(false, template)
+		})
+
+		AfterEach(func() {
+			err := deleteVirtualMachine(vm)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		table.DescribeTable("Should add volumes on an offline VM", func(addVolumeFunc func(name, namespace, volumeName, claimName, bus string), removeVolumeFunc func(name, namespace, volumeName string)) {
+			By("Adding test volumes")
+			addVolumeFunc(vm.Name, vm.Namespace, "some-new-volume1", "madeup", "scsi")
+			addVolumeFunc(vm.Name, vm.Namespace, "some-new-volume2", "madeup", "scsi")
+			By("Verifying the volumes have been added to the template spec")
+			verifyVolumeAndDiskVMAdded(vm, "some-new-volume1", "some-new-volume2")
+			By("Removing new volumes from VM")
+			removeVolumeFunc(vm.Name, vm.Namespace, "some-new-volume1")
+			removeVolumeFunc(vm.Name, vm.Namespace, "some-new-volume2")
+
+			verifyVolumeAndDiskVMRemoved(vm, "some-new-volume1", "some-new-volume2")
+		},
+			table.Entry("with DataVolume", addDVVolumeVM, removeVolumeVM),
+			table.Entry("with PersistentVolume", addPVCVolumeVM, removeVolumeVM),
+		)
+	})
+
+	Context("rook-ceph", func() {
```

Comment:
roman's going to kill me here... but for the time being, in order to ensure test consistency, I'd add `Context([Serial] rook-ceph)`. That ensures these tests will never be executed in parallel with other tests.

After we get this merged, id like to remove the Serial marker, but have that a separate PR.

Reply:
Added the serial marker.